File: app/src/main/java/com/psliusar/layers/sample/screen/fragment/FragmentLayer.java
package com.psliusar.layers.sample.screen.fragment;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.psliusar.layers.LayersFragment;
import com.psliusar.layers.Transition;
import com.psliusar.layers.sample.R;
import com.psliusar.layers.sample.screen.child.ChildLayer;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

/**
 * Logic is copied from {@link com.psliusar.layers.sample.screen.child.ChildrenContainerLayer}.
 */
public class FragmentLayer extends LayersFragment {

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.screen_children_container, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle state) {
        super.onViewCreated(view, state);
        if (state == null) {
            getLayers().at(R.id.children_container_top)
                .add(ChildLayer.class)
                .prepareLayer(new Transition.OnLayerTransition<ChildLayer>() {
                    @Override
                    public void onBeforeTransition(@NonNull ChildLayer layer) {
                        layer.setParameters("Top layer");
                    }
                })
                .setName("Top")
                .commit();

            getLayers().at(R.id.children_container_middle)
                .add(ChildLayer.class)
                .prepareLayer(new Transition.OnLayerTransition<ChildLayer>() {
                    @Override
                    public void onBeforeTransition(@NonNull ChildLayer layer) {
                        layer.setParameters("Middle layer");
                    }
                })
                .setName("Middle")
                .commit();

            getLayers().at(R.id.children_container_bottom)
                .add(ChildLayer.class)
                .prepareLayer(new Transition.OnLayerTransition<ChildLayer>() {
                    @Override
                    public void onBeforeTransition(@NonNull ChildLayer layer) {
                        layer.setParameters("Bottom layer");
                    }
                })
                .setName("Bottom")
                .commit();
        }

        view.findViewById(R.id.children_container_add_layer).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                final int number = getLayers().getStackSize() + 1;
                getLayers()
                    .add(ChildLayer.class)
                    .prepareLayer(new Transition.OnLayerTransition<ChildLayer>() {
                        @Override
                        public void onBeforeTransition(@NonNull ChildLayer layer) {
                            layer.setParameters("Stack layer " + number);
                        }
                    })
                    .setName("Stack #" + number)
                    .setOpaque(false)
                    .commit();
            }
        });
    }

    @Override
    protected int getDefaultContainerId() {
        return R.id.children_container;
    }
}

File: app/src/main/java/com/psliusar/layers/sample/screen/fragment/FragmentLayer.kt
package com.psliusar.layers.sample.screen.fragment

import android.os.Bundle
import android.view.View
import com.psliusar.layers.LayersFragment
import com.psliusar.layers.sample.R
import com.psliusar.layers.sample.screen.child.ChildLayer

/**
 * Logic is copied from [com.psliusar.layers.sample.screen.child.ChildrenContainerLayer].
 */
class FragmentLayer : LayersFragment(R.layout.screen_children_container) {

    override val defaultContainerId: Int = R.id.children_container

    override fun onViewCreated(view: View, state: Bundle?) {
        super.onViewCreated(view, state)
        if (state == null) {
            layers.at(R.id.children_container_top)
                .add<ChildLayer> {
                    withLayer { setParameters("Top layer") }
                    name = "Top"
                }

            layers.at(R.id.children_container_middle)
                .add<ChildLayer> {
                    withLayer { setParameters("Middle layer") }
                    name = "Middle"
                }

            layers.at(R.id.children_container_bottom)
                .add<ChildLayer> {
                    withLayer { setParameters("Bottom layer") }
                    name = "Bottom"
                }
        }

        getView<View>(R.id.children_container_add_layer).setOnClickListener {
            val number = layers.stackSize + 1
            layers.add<ChildLayer> {
                withLayer { setParameters("Stack layer $number") }
                name = "Stack #$number"
                opaque = false
            }
        }
    }
}
